controllers/disruption_controller.go

```diff
@@ -611,18 +610,19 @@ func (r *DisruptionReconciler) selectTargets(instance *chaosv1beta1.Disruption)
 	// won't pick up another one
 	targetsCount -= len(instance.Status.IgnoredTargets)
 
-	// prune ignored targets from all targets
-	for _, target := range matchingTargets {
-		if !contains(instance.Status.IgnoredTargets, target) {
-			eligibleTargets = append(eligibleTargets, target)
-		}
+	// filter matching targets to only get eligible ones
+	eligibleTargets, err := r.getEligibleTargets(instance, matchingTargets)
+	if err != nil {
+		r.log.Errorw("error getting eligible targets", "error", err)
+
+		return fmt.Errorf("error getting eligible targets: %w", err)
 	}
 
 	// if the asked targets count is greater than the amount of found targets, we take all of them
 	targetsCount = int(math.Min(float64(targetsCount), float64(len(eligibleTargets))))
 	if targetsCount < 1 {
 		r.log.Info("ignored targets has reached target count, skipping")
-		r.Recorder.Event(instance, "Warning", "NoTarget", "Ignored Targets is too large, we will no longer choose additional eligible targets for injection for this disruption")
+		r.Recorder.Event(instance, "Warning", "NoTarget", "No eligible targets found for injection for this disruption (either ignored or already targeted by another disruption)")
```

Comment:
This message is a little misleading? It can absolutely be the  case that we find many eligible targets but still emit this event.

Consider the situation where there are 15 possible targets, and our `count` is set to 5. If ignoredTargets has reached 5 (due to previously injected but now deleted target pods), even if all 15 possible targets are eligible, we still won't select any, and we'll emit this event

Reply:
I changed the message a bit, there's also a precision at the end of the message which could help understanding it? Otherwise we can split the logic in 2 different events, one when `ignored targets > count` and one when there's no eligible targets?